### ISSUE TITLE
Add optional "change" function to AccessibleValueHandler

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/sun#589]
This "change" function would behave similarly to Slider.js optional `drag` function.

We have an optional `startChange` and `endChange` but not a callback for every change. This would be useful for press and hold type functionality while using a keybaord, and would be beneficial in cases like https://github.com/phetsims/wave-interference/issues/484#issuecomment-614151459.

It hasn't been added yet because there is no concept of "press and hold" when using mobile or switch devices for accessibility. We only get single `change` or `input` events every user input.  But I propose we add a `change` function that behaves like

- Called once every `keydown` event before any other modification is made on the `valueProperty` (this is the behavior in Slider.js)
- Called once immediately after `startChange()` so that we still receive the callback once every input/gesture on devices which don't have "press and hold" capabilities.

**Comment by zepumph** [comment]
I think that this is a good first step, but I worry about the difference in API (generally) between `drag` and `change`. I would recommend one of two things:

1. In AccessibleValueHandler, only call `change` when not calling start/endChange.
2. Keep committed logic in AccessibleValueHandler since it makes sense in terms of "change", but then in `AccessibleSlider` handle logic that makes sure that `options.drag` is only fired when not coming from the start or end input changes.

I like (2) better, because it keeps the same API as Slider.js for startDrag/endDrag/drag callbacks. That said it may be more challenging.

- [ ] as an aside, it looks like we need to add options.drag to the merge in AccessibleSlider. That would be the best place to document how the logic is either the same or different from the Slider.drag option.

I really like the progress thus far. Thanks for adding this. I'm not sure if this is ready for a review just yet! But hopefully this helps.

**Comment by zepumph** [comment]
(I shouldn't have commented yet, I see https://github.com/phetsims/sun/commit/1d917fccaf7d2e31c1a165cf2a6cb6bee0a375c2 now)

**Comment by jessegreenberg** [comment]
Thanks so much for taking a look, was just about to request feedback from you!

>  but I worry about the difference in API (generally)

At first I totally agreed with you, but after trying out the change I believe I it is correct as is. For pointer events, the callbacks behave like
- `down` event: `startDrag`
- `move` event: `drag`
- `up` event: `endDrag`

In order to capture all of this with alternative input, we need to keep as we are now because there often isn't anything like the `move` event from the PDOM. So its like
- `keydown` event: `startChange` (if first press), `change`
- `keyup` event: `endChange`

or if we get a single `input` or `change` DOM event, we have to do all three sequentially to have the same behavior as pointer-like events.
-  `input` event: `startChange`, `change`, `endChange`

If we only called `change` when not coming from input that triggered `startChange` and `endChange`, we would generally miss calls to the `change` callback.

**Comment by zepumph** [comment]
I understand and trust you. Is it possible to create a test case where we find that a callback passed to `drag` is strange and broken when passed as the `change` callback? I think it would be very benefitial to understand if there are shortcommings in certain situations and then to document how the API for AccessibleSlider behaves differently (even though it is named the same) as Slider.js. Again I think the place to do that is when defining the AccessibleSlider.drag option.

Does that make sense? Is it reasonable?

Let me know if you want to discuss more.

**Comment by jessegreenberg** [comment]
I reviewed the above commits and comments again. I think the concern in https://github.com/phetsims/sun/issues/589#issuecomment-614265318 is that we perhaps didn't think through all the potential problems of passing the `drag` listener from Slider as the `onChange` listener in AccessibleValueHandler. 

Looking at where optional `drag` is called in Slider, I think that the current implementation in AccessibleValueHandler is correct and calls the callback from PDOM input, which was missing before. But would be good to review again with @zepumph, maybe at next a11y dev meeting.

**Comment by jessegreenberg** [comment]
Discussed with @zepumph again - since this issue has been open we haven't encountered any problems with the change, and it is working as expected for existing sliders with description/alternative input. Perhaps there will be a slider in the future where passing the startDrag/drag/endDrag callbacks directly as startChange/change/endChange callbacks will cause a problem and we will address it then when we encounter that case.

We decided to add some documentation to clarify that some of the Slider options are passed directly to the AccessibleSlider trait and agreed this can be closed.
